libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -207,12 +209,14 @@ class MapboxRouteLineApi(
         private set
     private val trafficBackfillRoadClasses = CopyOnWriteArrayList<String>()
     private var alternativesDeviationOffset = mapOf<String, Double>()
+
     private val alternativelyStyleSegmentsNotInLegCache: LruCache<
-        CacheResultUtils.CacheResultKey2<
-            Int, List<RouteLineExpressionData>,
-            List<RouteLineExpressionData>
+        CacheResultUtils.CacheResultKey3<
+            Int, List<RouteLineExpressionData>, Int,
+            List<RouteLineExpressionData>,
             >,
-        List<RouteLineExpressionData>> by lazy { LruCache(2) }
+        List<RouteLineExpressionData>> by lazy { LruCache(4) }
```

Comment:
Why cache 4 elements? There can be more than 4 legs in the route. Do we need this LRU cache at all? Couldn't we store data for the current active leg index only? We calculate the data only on leg change anyway and we won't need data for other (especially already traversed legs) anymore, we shouldn't need to store it.